x-pack/legacy/plugins/lens/public/xy_visualization/xy_expression.tsx

```diff
@@ -235,6 +237,41 @@ export function XYChart({
         theme={chartTheme}
         rotation={shouldRotate ? 90 : 0}
         xDomain={xDomain}
+        onBrushEnd={(min: number, max: number) => {
+          // in the future we want to make it also for histogram
+          if (!xAxisColumn || xAxisColumn.meta?.type !== 'date_histogram') {
+            return;
+          }
+
+          const table = data.tables[layers[0].layerId];
+          const xAxisFieldName: string | undefined = xAxisColumn?.meta?.aggConfigParams?.field;
+          const timeFieldName = xDomain && xAxisFieldName;
+
+          // TODO: simplify the context structure: https://github.com/elastic/kibana/issues/62936
+          const context: EmbeddableVisTriggerContext = {
+            data: {
+              range: [moment(min).toISOString(), moment(max).toISOString()],
+              data: {
+                ordered: {
+                  date: xAxisColumn.formatHint?.id === 'date',
+                },
+                series: [
+                  {
+                    values: table.rows.map((row, rowIndex) => ({
+                      xRaw: {
+                        table,
+                        column: table.columns.findIndex(el => el.id === xAxisColumn?.id), // index of X accessor in the table:
```

Comment:
Still trying to avoid using the response object, use the configuration instead `el.id === layers[0].xAccessor`